File: Brute_Force/javascript/1107.js
const input = require('fs').readFileSync('/dev/stdin').toString().split('\n');
const n = Number(input[0]);
const len = input[0].length;
const m = Number(input[1]);
let brokenButton
if (m>0) brokenButton = input[2].split(' ');

let result = Math.abs(n-100);
let trackCnt;
if (m!= 10 && m != 0){
  for (let i = 0 ; i<=1000000 ; i++){
    numString = String(i);
    digitBool = true;
    for (let digit of numString){
      if (brokenButton.includes(digit)){
        digitBool = false;
        break;
      }
    }
    if (!digitBool) continue;
    trackCnt = Math.abs(i-n) + numString.length;
    result = Math.min(result,trackCnt);
    if (result <= len) break;
  }
}
if (m==0) result = Math.min(result,len);
console.log(result);

File: Brute_Force/python/1107.py
n = input()
m = int(input())
if m != 0:
    broken_button = set(input().split())

result = abs(int(n)-100)
if m != 10 and m != 0:
    for num in range(900000):
        digit_bool = True
        for digit in str(num):
            if digit in broken_button:
                digit_bool = False
                break
        if not digit_bool: continue
        track_cnt = len(str(num)) + abs(num-int(n))
        result = min(track_cnt,result)
if m == 0:
    result = min(result,len(n))
print(result)